CMakeLists.txt

```diff
@@ -686,7 +686,8 @@ option(USE_MAINTAINER_MODE
 )
 
 if (USE_MAINTAINER_MODE)
-  set(IRESEARCH_DEBUG ON)
+  add_definitions("-DIRESEARCH_DEBUG")
```

Comment:
Nit:
By setting the definition here, we set it globally, so it will leak into other modules (e.g. RocksDB). Although in other modules it will not do any harm, because the define name has an `IRESEARCH_` prefix.

Potential improvement:
I guess this definition mostly affects code in the `3rdParty/iresearch` subdirectory. But as the define is picked up in headers files in `3rdParty/iresearch` that we include from the ArangoDB code, we need it in ArangoDB too. 
We could set it for specific targets (e.g. some targets inside ArangoDB and iresearch) only.

But it should already work as it is now.

Reply:
I made it global because some Arango headers include code relying on `IRESEARCH_DEBUG` macro and there was a very subtle issue related to that.

Reply:
@jsteemann I think we can improve it by write in the iresearch submodule

```cmake
target_compile_definitions(some_base_iresearch_target PUBLIC -DIRESEARCH_DEBUG)
```

Reply:
But from the other side, if someone don't link against iresearch and use header only files from iresearch it will be broken